packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQueryBuilderTemplateQueryPanelContent.tsx

```diff
@@ -0,0 +1,192 @@
+            generateDataSpaceTemplateQueryCreatorRoute(
+              queryBuilderState.projectInfo.groupId,
+              queryBuilderState.projectInfo.artifactId,
+              queryBuilderState.projectInfo.versionId,
+              queryBuilderState.dataSpace.path,
+              template.title,
+            ),
+          ),
+        );
+      }
+    };
+
+    return (
+      <ClickAwayListener onClickAway={handleClose}>
+        <div>
+          <BasePopover
+            open={queryBuilderState.isTemplateQueryDialogOpen}
+            PaperProps={{
+              classes: {
+                root: '"query-builder__data-space__template-query-panel__container__root',
+              },
+            }}
+            className="query-builder__data-space__template-query-panel__container"
+            onClose={handleClose}
+            anchorEl={triggerElement}
+            anchorOrigin={{
+              vertical: 'bottom',
+              horizontal: 'left',
+            }}
+            transformOrigin={{
+              vertical: 'top',
+              horizontal: 'center',
+            }}
+          >
+            <div className="query-builder__data-space__template-query-panel">
+              <div className="query-builder__data-space__template-query-panel__header">
+                Curated Template Queries
+              </div>
+              {templateQueries.map((query) => (
+                <div
+                  key={query.title}
+                  className="query-builder__data-space__template-query-panel__query"
+                >
+                  <TagIcon className="query-builder__data-space__template-query-panel__query__icon" />
+                  <button
+                    className="query-builder__data-space__template-query-panel__query__entry"
+                    title="click to load template query"
+                    onClick={() => loadQuery(query)}
+                  >
+                    <div className="query-builder__data-space__template-query-panel__query__entry__content">
+                      <div className="query-builder__data-space__template-query-panel__query__entry__content__title">
+                        {query.title}
+                      </div>
+                      {query.description && (
+                        <div className="query-builder__data-space__template-query-panel__query__entry__content__description">
+                          {query.description}
+                        </div>
+                      )}
+                    </div>
+                  </button>
+                  <button
+                    className="query-builder__data-space__template-query-panel__query__share"
+                    title="Share..."
+                    onClick={() => shareTemplateQuery(query)}
+                  >
+                    <ShareIcon />
+                    <div className="query-builder__data-space__template-query-panel__query__share__label">
+                      Share
```

Comment:
is 'Share` the right name here. we are opening the template query? Do you think visit is better?